public/src/client/topic.js

```diff
@@ -197,10 +198,20 @@ define('forum/topic', [
 			}
 		}
 
-		var currentBookmark = localStorage.getItem('topic:' + ajaxify.data.tid + ':bookmark');
+		var bookmarkKey = 'topic:' + ajaxify.data.tid + ':bookmark';
+		var currentBookmark = ajaxify.data.bookmark || localStorage.getItem(bookmarkKey);
 
-		if (!currentBookmark || parseInt(postIndex, 10) >= parseInt(currentBookmark, 10)) {
-			localStorage.setItem('topic:' + ajaxify.data.tid + ':bookmark', postIndex);
+		if (!currentBookmark || parseInt(postIndex, 10) > parseInt(currentBookmark, 10)) {
+			if (app.user.uid) {
+				var data = {
+					'tid':  ajaxify.data.tid,
+					'uid': app.user.uid,
+					'postIndex': postIndex
+				}
+				socket.emit('topics.bookmark', data);
```

Comment:
Add a callback function to this `socket.emit` that updates `ajaxify.data.bookmark`, otherwise after setting a bookmark, scrolling in any direction causes many socket calls to be made (e.g. flooding)